### PR TITLE
removed all *.class parameters

### PR DESCRIPTION
**Pull request by fabpot**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | yes |
| Deprecations? | no |
| Tests pass? | no |
| Fixed tickets | #11881, #12822 |
| License | MIT |
| Doc PR | none |

### CODE REVIEW

**Comment by fabpot** [comment]
Here is the script I used:

```php
<?php

require_once __DIR__.'/vendor/autoload.php';

use Symfony\Component\Finder\Finder;

$finder = (new Finder())->name('*.xml')->in(__DIR__.'/src');
$values = array();
foreach ($finder as $file) {
    $contents = file_get_contents($file);

    $new = preg_replace_callback("{^ +<parameter key=\"([^\"]+)\.class\">([^<]+)</parameter>\n}m", function ($match) use (&$values) {
        $values['%'.$match[1].'.class%'] = $match[2];
    }, $contents);

    $new = preg_replace("{^ +<parameters>\s+</parameters>\n\n}sm", '', $new);
    $new = preg_replace("{</parameter>\s+</parameters>}s", "</parameter>\n    </parameters>", $new);
    $new = preg_replace("{<parameters>\s+    <parameter}s", "<parameters>\n        <parameter", $new);

    if ($new != $contents) {
        file_put_contents($file, $new);
    }
}

$finder = (new Finder())->in(__DIR__.'/src');
foreach ($finder as $file) {
    $contents = file_get_contents($file);

    $new = preg_replace_callback('{%[a-zA-Z0-9\._]+\.class%}', function ($match) use ($values) {
        return isset($values[$match[0]]) ? $values[$match[0]] : $match[0];
    }, $contents);

    if ($new != $contents) {
        file_put_contents($file, $new);
    }
}
```

**Comment by TomasVotruba** [comment]
:+1:

File: src/Symfony/Bundle/TwigBundle/Tests/DependencyInjection/TwigExtensionTest.php [hunk 1]
```diff
@@ -30,8 +30,6 @@ public function testLoadEmptyConfiguration()
         $container->loadFromExtension('twig', array());
         $this->compileContainer($container);
 
-        $this->assertEquals('Twig_Environment', $container->getParameter('twig.class'), '->load() loads the twig.xml file');
```
Comment (Tobion) [inline src/Symfony/Bundle/TwigBundle/Tests/DependencyInjection/TwigExtensionTest.php]: I guess this tests needs to be replaced by testing whether the `twig` service is defined to keep that test

File: src/Symfony/Bundle/TwigBundle/Tests/DependencyInjection/TwigExtensionTest.php [hunk 2]
```diff
@@ -51,8 +49,6 @@ public function testLoadFullConfiguration($format)
         $this->loadFromFile($container, 'full', $format);
         $this->compileContainer($container);
 
-        $this->assertEquals('Twig_Environment', $container->getParameter('twig.class'), '->load() loads the twig.xml file');
```
Comment (Tobion) [inline src/Symfony/Bundle/TwigBundle/Tests/DependencyInjection/TwigExtensionTest.php]: same

**Comment by Tobion** [comment]
Apart from the two small things looks ok to me.

File: src/Symfony/Bundle/TwigBundle/Tests/DependencyInjection/TwigExtensionTest.php (hunk 1 above)
Comment (fabpot) [inline src/Symfony/Bundle/TwigBundle/Tests/DependencyInjection/TwigExtensionTest.php]: fixed

File: src/Symfony/Bundle/TwigBundle/Tests/DependencyInjection/TwigExtensionTest.php (hunk 2 above)
Comment (fabpot) [inline src/Symfony/Bundle/TwigBundle/Tests/DependencyInjection/TwigExtensionTest.php]: fixed

**Comment by Tobion** [comment]
:+1:

**Pull request merged**